includes/Checker/Abstract_Check_Runner.php

```diff
@@ -270,11 +270,12 @@ final public function get_checks_to_run() {
 		// Include file to use is_plugin_active() in CLI context.
 		require_once ABSPATH . 'wp-admin/includes/plugin.php';
 
+		$runtime     = new Runtime_Environment_Setup();
```

Comment:
This breaks the [dependency inversion principle](https://en.wikipedia.org/wiki/Dependency_inversion_principle). In short, you should not hard code a class instantiation. This should be injected into the class. This helps mocking in tests. We do this in other places in the codebase.  Are we following the pattern of dependancy injection model in this plugin @felixarntz ?

Reply:
We do instantiate other classes in this class, but agree this does not align with how we approach this in other classes. I'm interested to hear @felixarntz thoughts on this for this case.

Reply:
Makes sense to allow setting this instance instead of hard-coding it, though we also should keep that optional, so that we don't have to set it everywhere where a runner is instantiated. I also wouldn't want to add it as a constructor parameter, since that would mean we have to maintain this forever or deprecate a parameter, which you can never truly replace later.

I would suggest we go with:
* A `private` class property for the `Runtime_Environment_Setup` instance
* Instantiate that property in the constructor with our regular instance
* Add a `public set_runtime_environment_setup( $runtime_environment_setup )` method that e.g. a test can optionally call to overwrite it.

Reply:
The suggestion is implemented in https://github.com/10up/plugin-check/pull/173/commits/8717db9cd7e7c87099df67324a70187716cd52ea